cpp/src/c_api/graph_functions.cpp

```diff
@@ -261,6 +261,74 @@ struct count_multi_edges_functor : public cugraph::c_api::abstract_functor {
   }
 };
 
+struct has_vertex_functor : public cugraph::c_api::abstract_functor {
+  raft::handle_t const& handle_{};
+  cugraph::c_api::cugraph_graph_t* graph_{nullptr};
+  const int vertex_{};
+  bool_t result_{bool_t::FALSE};
+  bool do_expensive_check_{false};
+
+  has_vertex_functor(::cugraph_resource_handle_t const* handle,
+                     ::cugraph_graph_t* graph,
+                     const int vertex,
+                     bool do_expensive_check)
+    : abstract_functor(),
+      handle_(*reinterpret_cast<cugraph::c_api::cugraph_resource_handle_t const*>(handle)->handle_),
+      graph_(reinterpret_cast<cugraph::c_api::cugraph_graph_t*>(graph)),
+      vertex_(vertex),
+      do_expensive_check_(do_expensive_check)
+  {
+  }
+
+  template <typename vertex_t,
+            typename edge_t,
+            typename weight_t,
+            typename edge_type_type_t,
+            bool store_transposed,
+            bool multi_gpu>
+  void operator()()
+  {
+    if constexpr (!cugraph::is_candidate<vertex_t, edge_t, weight_t>::value) {
+      unsupported();
+    } else {
+      auto graph =
+        reinterpret_cast<cugraph::graph_t<vertex_t, edge_t, false, multi_gpu>*>(graph_->graph_);
+
+      auto graph_view = graph->view();
+      auto number_map = reinterpret_cast<rmm::device_uvector<vertex_t>*>(graph_->number_map_);
+
+      rmm::device_uvector<vertex_t> vertex_array(1, handle_.get_stream());
+
+      cugraph::detail::sequence_fill(
+        handle_.get_stream(), vertex_array.data(), vertex_array.size(), vertex_t(vertex_));
+
+      if constexpr (multi_gpu) {
+        vertex_array = cugraph::shuffle_ext_vertices(handle_, std::move(vertex_array));
+      }
+
+      cugraph::renumber_ext_vertices<vertex_t, multi_gpu>(
+        handle_,
+        vertex_array.data(),
+        vertex_array.size(),
+        number_map->data(),
+        graph_view.local_vertex_partition_range_first(),
+        graph_view.local_vertex_partition_range_last(),
+        do_expensive_check_);
```

Comment:
I assume the same vertex_t value is passed to every GPU in multi-GPU, right? Or are you assuming that each GPU can call this function with different `vertex_` values?

Reply:
If the former is the case, you don't need to really shuffle vertices, just need to call `compute_gpu_id_from_ext_vertex_t` to find the owning GPU. The owning GPU performs the check and broadcast the check result.